### DISCUSSION TITLE
[Question][Hw1][Problem 7] How should we consider diverging case?

### DISCUSSION THREAD

**Discussion by yusungsim** [prosyslab-classroom/cs524-program-analysis#408] [Q&A]
Hi. I think that simplified version of PageRank algorithm has some diverging case so that iteration does not end.

For example, consider following `web` object
```ocaml
let web = [(0, [1]); (1, [2]); (2, [1]);]
```
In this case, the algorithm will infinitely loop between two imporance states, `(0: 0, 1: 2, 2: 1)` and `(0: 0, 1: 1, 2: 2)`.

Is my understanding correct? If yes, what should we do about this case?

(I think if all pages have self-referencing links, there will be no diverging case...)

**Comment by KihongHeo** [comment]
In this homework, let us ignore divergent cases.

**Comment by hyunsukimsokcho** [comment]
Closing.. please feel free to re-open if needed.
